src/serde/de.rs

```diff
@@ -321,7 +308,9 @@ where
                 std::mem::size_of::<V::Value>(),
                 std::mem::size_of::<Decimal>()
             );
+            // # Safety
             // compiler doesn't understand that the generic Decimal here is actually V::Value here
+            // the assert statement above that compares sizes of both the Decimal and V::Value ensures the conversion performed below is safe
```

Comment:
I'd avoid saying that the check ensures the conversion is safe; we don't have a way to know for a fact that `V::Value` is actually a `Decimal`.

```suggestion
            // The assert statement above that compares the sizes of the Decimal and V::Value types
            // as a safeguard. If a project actually had a newtype struct with the unlikely name of 
            // `$__ion_rs_decimal__`, it would be possible for `V::Value` to be that type instead.
            // We cannot confirm the actual type, so we confirm the expected size as a proxy.
```